src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -523,6 +539,11 @@ public static Vector256<uint> Insert(Vector256<uint> value, uint data, byte inde
         /// </summary>
         public static Vector256<long> Insert(Vector256<long> value, long data, byte index)
         {
+            if (IntPtr.Size != 8)
```

Comment:
So with this change, all `long/ulong` operations will throw PNSE when running in a 32-bit process?

Reply:
@eerhardt, not quite.

There are two basic types of operations:
* Those that take/return only `Vector128<long>` or `Vector128<ulong>`
  * These are fine in both 32-bit and 64-bit mode, as they operation on 128-bit registers (which exist in both 32-bit and 64-bit modes), on 64-bit memory (Scalar `Vector128<T>`) , or on 128/256-bit memory (Packed `Vector128<T>` and Packed `Vector256<T>`)
* Those that take/return `long` or `ulong` directly, which falls into two-subcategories
  * Those that require a 64-bit register and cannot work on 32-bit
    * `Sse2.StoreNonTemporal` and `Sse2.ConvertToInt64(Vector128<double>)` are examples that don't work, as they require at least one 64-bit register
  * Those that can operate directly on 64-bit memory, and don't require a register
    * These sometimes work on 32-bit machines
    * `ConvertToInt64(Vector128<long>)` is an example that, as it uses a special encoding of the `movq` instruction
    * `ConvertScalarToVector128Double(Vector128<double>, long)` is an example of one that supports taking just a 64-bit memory address, but is still not encodable on a 32-bit machine

Reply:
It seems to me that we shouldn't do anything tantamount to emulation when executing on a 32-bit machine. That said, I'm wondering if there are suggestions as to how these limitations/exclusions are reflected in the usage model?

Reply:
Arm32/64 is differentiating by the `Arm` and `Arm.Arm64` namespaces (however, this is also partially due to these being entirely different architectures, rather than hierarchical, like x86/x64).

Exposing a new class would probably work, but would also break the `make Sse2 inherit from Sse` proposal since we don't have multiple inheritance or interfaces that can declare static methods.

Duplicating all the members under a separate `x64` namespace would also work, but that is a lot of duplicated metadata for relatively few types.

I had also thought of just updating the reference assembly to have a x86/x64 specific version; but we are exposing all architectures from a single package so the cross-architecture story is nicer (and so you don't need `#ifdef`)

Reply:
Relying on an analyzer/documentation to help notify users of these incompatibilities seems like it might the easiest way to do this...

Reply:
https://github.com/dotnet/coreclr/issues/18744